### PR TITLE
Modernize attendance tool backend

### PR DESCRIPTION
**Pull request by davidpbauer**
Migrate from hbm xml to JPA annotations
Reduce bloat from data model
Reorganize api packages
Remove stats from the database
Remove stats calc job
Add ignite caches

### CODE REVIEW

**Comment by davidpbauer** [comment]
Ignite contrib caches for Sakai 23 - https://github.com/sakaiproject/sakai/pull/10808

**Comment by davidpbauer** [comment]
This is working well for me locally now. I'm not sure why the maven build is failing on github...

Can someone else take a look at these changes and let me know what you think?

I plan to move on to updating the UI to better match the rest of Sakai once this is merged.

**Comment by ottenhoff** [comment]
do you need to have sakai kernel in pom.xml?

File: api/src/java/org/sakaiproject/attendance/api/model/AttendanceEvent.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        return getClass().hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "(" +
+                "id = " + id + ", " +
+                "name = " + name + ", " +
+                "startDateTime = " + startDateTime + ")";
+    }
```
Comment (ern) [inline api/src/java/org/sakaiproject/attendance/api/model/AttendanceEvent.java:99]: To use lombok to generate these overridden implementations

**Comment by ottenhoff** [comment]
Is this PR still relevant?

**Comment by ottenhoff** [comment]
These changes all look good @davidpbauer .... can we grab what you're running from a UD branch?

**Comment by davidpbauer** [comment]
Hi Sam. This work never made it to production at UD due to the timing of it, but this branch now has the most up-to-date version of our changes and is testing well locally.

**Pull request closed without merging**